### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vadivel-Babu**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -20,6 +20,17 @@ function calculateMenAverageAge(people, century) {
...
   // without nesting
+  const men = people.filter(
+    (person) =>
+      person.sex === 'm'
+      && (century === Math.ceil(person.died / 100) || !century)
+  );
+
+  const averageAge
+    = men.map((man) => man.died - man.born).reduce((a, b) => a + b, 0)
+    / men.length;
```
Comment (qoa11a) [inline src/averageAges.js:31]: Create a function to calculate averageAge as you use it more than one time.

**Review by qoa11a — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 2]
```diff
@@ -14,12 +14,30 @@
  *
  * @return {number}
  */
+
+function callBack(arr, len) {
```
Comment (polosanya) [inline src/averageAges.js:18]:
```suggestion
function calculateAverageAge(ages) {
```
Let's use more descriptive and meaningful names. 
Also, why do you need to pass the array length? You can get it inside the function

File: src/averageAges.js [hunk 3]
```diff
@@ -38,6 +56,19 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people
+    .filter(
+      (person) =>
+        person.sex === 'f'
+        && (withChildren
```
Comment (polosanya) [inline src/averageAges.js:63]:
```suggestion
        person.sex === 'f'
          && (withChildren
```
Fix all cases

File: src/averageAges.js [hunk 4]
```diff
@@ -38,6 +56,19 @@ function calculateMenAverageAge(people, century) {
...
+          ? people.find((child) => child.mother === person.name)
+          : !withChildren)
+    )
+    .map((woman) => woman.died - woman.born);
+
+  const result = callBack(women, women.length);
```
Comment (polosanya) [inline src/averageAges.js:69]: Result is not the best name for variable. Try to specify what you get from this function call.
Fix all cases

**Review by polosanya — CHANGES_REQUESTED** [review]
Almost perfect!

File: src/averageAges.js (hunk 3 above)
Comment (polosanya) [reply src/averageAges.js:63]: Not fixed

File: src/averageAges.js [hunk 5]
```diff
@@ -56,6 +87,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people
+    .filter(
+      (person) =>
+        people.some((mom) => mom.name === person.mother)
+        && (onlyWithSon ? person.sex === 'm' : !onlyWithSon)
```
Comment (polosanya) [inline src/averageAges.js:94]:
```suggestion
        people.some((mom) => mom.name === person.mother)
          && (onlyWithSon ? person.sex === 'm' : !onlyWithSon)
```

**Review by polosanya — APPROVED** [review]
Good job!

**Pull request closed without merging**